flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/internal/TransactionAbortStrategyImpl.java

```diff
@@ -107,6 +119,64 @@ private int abortTransactionOfSubtask(String prefix, int subtaskId, Context cont
             }
             return numTransactionAborted;
         }
+    },
+    LISTING {
+        private final EnumSet<TransactionState> abortableStates =
+                EnumSet.complementOf(
+                        EnumSet.of(
+                                TransactionState.COMPLETE_ABORT,
+                                TransactionState.COMPLETE_COMMIT,
+                                TransactionState.PREPARE_COMMIT));
+
+        @Override
+        public void abortTransactions(Context context) {
+
+            Collection<TransactionListing> openTransactionsForTopics =
+                    AdminUtils.getOpenTransactionsForTopics(
+                            context.getAdminClient(), context.getTopicNames());
+
+            // This list could be huge if TransactionNamingStrategy#INCREMENTING is used, so cap it
+            LOG.trace(
```

Comment:
Let's warn the user how many transactions will be aborted to spot unwanted behavior earlier

Reply:
+1 this should be `WARN`

Reply:
I usually only use WARN for things that are actionable and would prefer INFO instead. But I'll abide by the majority here ;). (I agree that trace is way too low)